### DISCUSSION TITLE
Support for Video Thumbnails on Upload

### DISCUSSION THREAD

**Discussion by boldeuler** [pocketbase/pocketbase#5835] [Q&A] [answered]
Hi everyone,

Firstly, thank you for the great work on PocketBase and its robust feature set, including the existing support for image thumbnails.

I wanted to ask if there are any plans or current capabilities to extend thumbnail generation to video files upon upload. This would be highly beneficial for applications where video previews are necessary.

Questions:

Are there any technical challenges or considerations that have been evaluated regarding video thumbnail generation?

If this feature isn't currently supported, would there be recommendations for implementing this functionality as an extension or integration?

I look forward to any insights you might have on this topic!

Best regards

**Comment by ganigeorgiev** [comment, marked as answer]
At least for the time being there are no plans to generate thumbs for videos out of the box as it could be very resource intensive and I'm not sure if it would be useful enough to worth the troubles developing and maintaining it.

One possible workaround could be to capture a screenshot from the video on the client-side before upload and send it together with the video file as separate image `file` field.

Keep in mind that PocketBase is not well tested or optimized for uploading and serving very large files (e.g. 1GB+) and I don't feel confident recommending it for such cases. Based on my personal experience, video encoding and delivery is rarely suitable for applications like PocketBase that can't scale on more than 1 server.

**Reply by boldeuler** [reply]
Thanks for your great workaround idea! I will apply it. Current I support 50MB files only.

**Comment by boldeuler** [comment]
While implementing the accepted answer I found another solution

```
<video
            [autoplay]="false"
            [controls]="false"
            [src]="src + '#t=1.5'"
            [disableRemotePlayback]="true"
            [muted]="true"
            preload="metadata"
            ></video>
```
This loads only the first 1.5 seconds of the video.


It may load the whole video if the nginx header is not set.
```
        proxy_set_header Range $http_range;
```

**Reply by VictorioBerra** [reply]
Yes that is what I do. Browsers do it automatically now. Alternatively I have also used a hook to run ffmpeg to extract a random frame to use as a thumbnail.

**Comment by areeb111** [comment]
The most recommended approach, Is to have a ffmpeg instance on the server, generate the thumbnail, and push it to a new image field.
That would be more faster for the client while browsing.
